### PR TITLE
Template version 1.0.0

### PR DESCRIPTION
**Pull request by FlatBallFlyer**
Version 1.0 - Unit test coverage is so-so, Step-CI testing is so-so - but it works.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,37 +1,39 @@
-# Base build stage
-FROM python:3.10 AS base
-  
-WORKDIR /src
-COPY . .
+# Stage 1: Build stage
+FROM python:3.12-slim as build
+
+# Install Git and other dependencies
+RUN apt-get update && apt-get install -y git
```
Comment (michquinn) [inline Dockerfile:5]: Where is this being used?

File: README.md [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
+- [ ] [MongoIO](./src/utils/mongo_io.py)
+- [ ] [server.py](/src/server.py)
+- [ ] [All Test Files](./test/)
+
+Now you can remove these instructions from the readme, 
 
 ## Overview
 
-This is a simple Flask API that provides Get/Post/Patch services for docuements in the Encounter collection. This API uses data from a [backing Mongo Database](https://github.com/agile-learning-institute/mentorHub-mongodb), and supports a [Single Page Application.](https://github.com/agile-learning-institute/mentorHub-encounter-ui)
+This is a template repo for a simple Flask API that provides Get/Post/Patch services for documents in the encounter encounter. This API uses data from a [backing Mongo Database](https://github.com/agile-learning-institute/mentorHub-mongodb), and sup8090s a [Single Page Application.](https://github.com/agile-learning-institute/mentorHub-encounter-ui)
```
Comment (michquinn) [inline README.md:13]:
```suggestion
This is a template repo for a simple Flask API that provides Get/Post/Patch services for documents in the encounter collection. This API uses data from a [backing Mongo Database](https://github.com/agile-learning-institute/mentorHub-mongodb), and supports a [Single Page Application.](https://github.com/agile-learning-institute/mentorHub-encounter-ui)
```

File: README.md [hunk 3]
```diff
@@ -16,124 +24,140 @@ The OpenAPI specifications for the api can be found in the ``docs`` folder, and
...
 
-## API Testing with CURL
+# API Testing with CURL
 
 If you want to do more manual testing, here are the curl commands to use
 
 ### Test Health Endpoint
 
-This endpoint supports the promethius monitoring standards for a healthcheck endpoint
+This endpoint sup8090s the Prometheus monitoring standards for a health check endpoint
```
Comment (michquinn) [inline README.md:89]:
```suggestion
This endpoint supports the Prometheus monitoring standards for a health check endpoint
```

File: README.md [hunk 4]
```diff
@@ -16,124 +24,140 @@ The OpenAPI specifications for the api can be found in the ``docs`` folder, and
...
-### Test get all encounters
+### Test get a Encounter document
 
 ```bash
-curl http://localhost:8090/api/encounters/
+curl http://localhost:8088/api/encounter/eeee00000000000000000001/
 ```
 
-### Test add a new encounter
+### Test add a Encounter Document
```
Comment (michquinn) [inline README.md:107]:
```suggestion
### Test add an Encounter Document 
```

File: README.md [hunk 5]
```diff
@@ -16,124 +24,140 @@ The OpenAPI specifications for the api can be found in the ``docs`` folder, and
...
-curl -X POST http://localhost:8090/api/encounters/ \
-     -d '{"personId":"aaaa00000000000000000000", "mentorId":"aaaa00000000000000000000"}'
-
+curl -X POST localhost:8088/api/encounter/ \
+-H "Content-Type: application/json" \
+-d '{"date":"2024-12-07 00:00:00","mentorId":"aaaa00000000000000000005","personId":"aaaa00000000000000000024","planId":"eeff00000000000000000004","status":"Active"}'
 ```
 
-### Test update an encounter
+### Test update a Encounter Document
```
Comment (michquinn) [inline README.md:115]:
```suggestion
### Test update an Encounter Document
```

File: README.md [hunk 6]
```diff
@@ -16,124 +24,140 @@ The OpenAPI specifications for the api can be found in the ``docs`` folder, and
...
+The ```api/config/``` endpoint will return a list of configuration values. These values are either "defaults" or loaded from a singleton configuration file, or an Environment Variable of the same name. Configuration files take precedence over environment variables. The environment variable "CONFIG_FOLDER" will change the location of configuration files from the default of ```./```
 
-The ```api/health/``` endpoint is a Promethius Healthcheck endpoint.
+The ```api/health/``` endpoint is a [Prometheus](https://prometheus.io) Health check endpoint.
 
-The [Dockerfile](./Dockerfile) uses a 2-stage build, and supports both amd64 and arm64 architectures. See [docker-build.sh](./src/docker/docker-build.sh) for details about how to build in the local architecture for testing, and [docker-push.sh] for details about how to build and push multi-architecture images.
+The [Dockerfile](./Dockerfile) uses a 2-stage build, and sup8090s both amd64 and arm64 architectures.
```
Comment (michquinn) [inline README.md:163]:
```suggestion
The [Dockerfile](./Dockerfile) uses a 2-stage build, and supports both amd64 and arm64 architectures. 
```

File: docs/openapi.yaml [hunk 7]
```diff
@@ -1,210 +1,198 @@
...
-      description: Creates a new encounter between a mentor and a member
-      operationId: addEncounter
-      requestBody:
-        description: Details of the encounter
-        content:
-          application/json:
-            schema:
-              $ref: '#/components/schemas/encounters'
-        required: true
+      summary: Create a encounter document
```
Comment (michquinn) [inline docs/openapi.yaml:15]:
```suggestion
      summary: Create an encounter document
```

File: docs/openapi.yaml [hunk 8]
```diff
@@ -1,210 +1,198 @@
...
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/encounters'
-  /api/encounters/{id}/:
-    patch:
-      summary: Update an existing encounter
-      description: Updates an existing encounter with the provided ID
-      operationId: updateEncounter 
+      summary: Get a encounter document
```
Comment (michquinn) [inline docs/openapi.yaml:29]:
```suggestion
      summary: Get an encounter document
```

File: docs/openapi.yaml [hunk 9]
```diff
@@ -1,210 +1,198 @@
...
-          description: Validation exception
-    get:
-      summary: Get an existing encounter
-      description: Retrieves an existing encounter with the provided ID
-      operationId: getEncounter
+                $ref: '#/components/schemas/encounter'
+        '500':
+          description: A Processing Error occurred
+    patch:
+      summary: Update a encounter document
```
Comment (michquinn) [inline docs/openapi.yaml:50]:
```suggestion
      summary: Update an encounter document
```

**Review by michquinn — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (FlatBallFlyer) [reply Dockerfile:5]: git rev-parse in
```
RUN BRANCH=$(git rev-parse --abbrev-ref HEAD) && \
    DATE=$(date "+%Y-%m-%d:%H:%M:%S") && \
    echo $DATE.$BRANCH > /src/PATCH_LEVEL
```

**Pull request merged**